nodejs/eks/cluster.ts

```diff
@@ -314,22 +314,40 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
 
     const roleMappings = pulumi.all([pulumi.output(args.roleMappings || []), instanceRoleMappings])
         .apply(([mappings, instanceMappings]) => {
-            return jsyaml.safeDump([...mappings, ...instanceMappings].map(m => ({
+            mappings.map(m => {
+                if (!checkArnExists(pulumi.output(m.roleArn))) {
+                    throw new Error(`The IAM role ARN ${m.roleArn} could not be properly retrieved or does not exist`);
+                }
+            });
+            const mappingYaml = jsyaml.safeDump([...mappings, ...instanceMappings].map(m => ({
                 rolearn: m.roleArn,
                 username: m.username,
                 groups: m.groups,
             })));
+            if (!mappingYaml) {
+                throw new Error(`The IAM role mappings provided could not be properly serialized to YAML for the aws-auth ConfigMap`);
+            }
```

Comment:
From the linked issue - doesn't `safeDump` throw?  If so, don't we need to catch the internal `sameDump` error and provider a better one?

Reply:
Done